listener.go

```diff
@@ -72,15 +75,26 @@ func (c *Conn) Listener(port uint16, key crypto.PrivateKey) (net.Listener, error
 		return nil, newProtocolError("failed to extract local port")
 	}
 
+	if len(vports) < 1 {
+		return nil, newProtocolError("no virual ports specified")
+	}
+	targetPortStr := strconv.FormatUint((uint64)(tAddr.Port), 10)
+	var portSpecs []OnionPortSpec
+	for _, vport := range vports {
+		portSpecs = append(portSpecs, OnionPortSpec{
+			VirtPort: vport,
+			Target:   targetPortStr,
+		})
+	}
+	cfg.PortSpecs = portSpecs
```

Comment:
Okay, I'll make a copy of this config (though it seems to be a bit overkill) and add a note about `cfg.PortSpecs`.